h2/utilities.py

```diff
@@ -147,7 +148,11 @@ def authority_from_headers(headers):
     return None
 
 
-def validate_headers(headers):
+ConnectionState = collections.namedtuple('ConnectionState',
+                                         ['is_client', 'is_trailer'])
```

Comment:
Two hard problems in computer science, etc. :smile: 

(Also, it turns out there already ~is something called `ConnectionState` in hyper. What could possibly go wrong from a second object with the same name?)

Fixed.